components/camel-sql/src/main/java/org/apache/camel/processor/aggregate/jdbc/JdbcCamelCodec.java

```diff
@@ -69,18 +81,13 @@ public Exchange unmarshallExchange(CamelContext camelContext, byte[] buffer) thr
         return answer;
     }
 
-    private byte[] encode(Object object) throws IOException {
-        ByteArrayOutputStream bytesOut = new ByteArrayOutputStream();
-        ObjectOutputStream objectOut = new ObjectOutputStream(bytesOut);
-        objectOut.writeObject(object);
-        objectOut.close();
-        byte[] data = bytesOut.toByteArray();
-        return data;
+    private void encode(Object object, OutputStream bytesOut) throws IOException {
+        try (ObjectOutputStream objectOut = new ObjectOutputStream(bytesOut)) {
+            objectOut.writeObject(object);
+        }
     }
 
-    private DefaultExchangeHolder decode(CamelContext camelContext, byte[] dataIn) throws IOException, ClassNotFoundException {
-        ByteArrayInputStream bytesIn = new ByteArrayInputStream(dataIn);
-
+    private DefaultExchangeHolder decode(CamelContext camelContext, InputStream bytesIn) throws IOException, ClassNotFoundException {
         ObjectInputStream objectIn = null;
```

Comment:
I avoided the urge to reduce this method to try-with-resources in case people see deeper meaning in using IOHelper#close ...